packages/core/strapi/lib/commands/opt-out-telemetry.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const { resolve } = require('path');
+const fse = require('fs-extra');
+const chalk = require('chalk');
+const fetch = require('node-fetch');
+const machineID = require('../utils/machine-id');
+
+const readPackageJSON = async path => {
+  try {
+    const packageObj = await fse.readJson(path);
+    const uuid = packageObj.strapi.uuid;
+
+    return [uuid, packageObj];
+  } catch (err) {
+    console.error(`${chalk.red('Error')}: ${err.message}`);
+  }
+};
+
+const writePackageJSON = async (path, file, spacing) => {
+  try {
+    await fse.writeJson(path, file, { spaces: spacing });
+    return true;
+  } catch (err) {
+    console.error(`${chalk.red('Error')}: ${err.message}`);
+  }
+};
+
+const sendEvent = async uuid => {
+  try {
+    await fetch('https://analytics.strapi.io/track', {
+      method: 'POST',
+      body: JSON.stringify({
+        event: 'didOptOutTelemetry',
+        uuid,
+        deviceId: machineID(),
+      }),
+      headers: { 'Content-Type': 'application/json' },
+    });
+  } catch (e) {
+    //...
+  }
+};
+
+module.exports = async function optOutTelemetry() {
+  const packageJSON = resolve(process.cwd(), 'package.json');
+
+  if (!packageJSON) {
```

Comment:
The result of the `resolve` function will always return something (even if the package.json doesn't exist) since it just processes path arguments.

So what you are trying to do here won't work, this condition will if, I'm not mistaken, never be run.

By the way, you can remove the `process.cwd()` call just above since you didn't prefix `package.json` with a `/`, `resolve` will try to resolve the final path from the current working directory by default.

Reply:
I chose `process.cwd()` here because for me it just looks a bit more readable in accordance with their documentations, that all.
Regarding the check for packageJSON - it's there only to check if for some reason the `resolve()` itself malfunctions. But there are other things up the chain which check if command is run from a strapi project directory and if package.json is in place. Just a precaution against `resolve()` malfunctioning for some reason and returning nothing.